src/python/pants/option/global_options.py

```diff
@@ -286,8 +286,9 @@ def register_bootstrap_options(cls, register):
                   'The `--pants-distdir` and `--pants-workdir` locations are inherently ignored.')
     register('--glob-expansion-failure', advanced=True,
              default=GlobMatchErrorBehavior.warn, type=GlobMatchErrorBehavior,
-             help="Raise an exception if any targets declaring source files "
-                  "fail to match any glob provided in the 'sources' argument.")
+             help="What to do when files specified on the command line or in BUILD files cannot be "
```

Comment:
Hm... see comment on #9009. I'm not sure it's a good idea to overload this. The CLI feels like it should always fail, period.

Reply:
I'm not sure I agree. A warning is a nicer user experience than an error IMO

If I run `./pants cloc '*.py' '*.java'`, and it turns out I only have Python files, it will clearly output in the warning that I don't have Java and still give me meaningful results for Python files. I, as a user, can then decide "hm I do really want to know Java results, let me rerun this" or "okay cool, TIL that we don't have Java".

cc @codealchemy (not at all for you to need to choose a particular side, only because I think you have a great eye for UX)

Reply:
(Me merging this PR is in no way trying to shut down this conversation early. I plan to have multiple follow up PRs that touch path globs and this option, and am happy to make the change in a followup if we decide to go that route. I only want to merge this now so that I can start putting up followup PRs.)

Reply:
If I'm following things here right, I partial to sticking with a warning as well - it helps guide a user (the followup change to be more specific will be particularly helpful there) to better use the tool without blocking what they may be trying to do (as a failure would).